src/profile/Signup.js

```diff
@@ -12,93 +15,86 @@ import LoaderButton from "../components/LoaderButton";
  * This component signs you up for the Cognito user
  * @TODO Issue #7
  * 
- * import useState
- *    isLoading: false,
-      email: "",
-      password: "",
-      confirmPassword: "",
-      confirmationCode: "",
-      newUser: null,
+ * import useState x
+ *    isLoading: false, x
+      email: "", x 
+      password: "", x
+      confirmPassword: "", x
+      confirmationCode: "", x
+      newUser: null, x
 
       bind forms to several useStates
  * 
- * import useHistory
+ * import useHistory x 
  * history.push("/onboarding/user");
  * 
  * 
  
  */
 
-export default class Signup extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      isLoading: false,
-      email: "",
-      password: "",
-      confirmPassword: "",
-      confirmationCode: "",
-      newUser: null,
-    };
-  }
+      
+const Signup = () => {
+  const [email, setEmail] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmationCode, setConfirmationCode] = useState('');
+  const [newUser, setNewUser] = useState(null);
+
 
-  validateForm() {
+  // for redirect to new route
+  const history = useHistory()
+
+  const validateForm = () => {
     return (
-      this.state.email.length > 0 &&
-      this.state.password.length > 0 &&
-      this.state.password === this.state.confirmPassword
+      email.length > 0 &&
+      password.length > 0 &&
+      password === confirmPassword
     );
   }
 
-  validateConfirmationForm() {
-    return this.state.confirmationCode.length > 0;
+  const validateConfirmationForm = () => {
+    return confirmationCode.length > 0;
   }
 
-  handleChange = (event) => {
-    this.setState({
-      [event.target.id]: event.target.value,
-    });
-  };
-
-  handleSubmit = async (event) => {
+  const handleConfirmationSubmit = async (event) => {
     event.preventDefault();
 
-    this.setState({ isLoading: true });
+    setIsLoading(true)
 
     try {
-      const newUser = await Auth.signUp({
-        username: this.state.email,
-        password: this.state.password,
-      });
-      this.setState({
-        newUser,
-      });
+      await Auth.confirmSignUp(email, confirmationCode);
+      await Auth.signIn(email, password);
+      history.push("/onboarding/user");
     } catch (e) {
       alert(e.message);
+      setIsLoading(false)
     }
 
-    this.setState({ isLoading: false });
   };
 
-  handleConfirmationSubmit = async (event) => {
+
+  const handleSubmit = async (event) => {
     event.preventDefault();
 
-    this.setState({ isLoading: true });
+    setIsLoading(true)
 
     try {
-      await Auth.confirmSignUp(this.state.email, this.state.confirmationCode);
-      await Auth.signIn(this.state.email, this.state.password);
-      this.props.history.push("/onboarding/user");
+      const newUser = await Auth.signUp({
```

Comment:
Can you please install Prettier on VS Code (assuming you are using VS Code), and can you run the linting command (auto-formatting, makes the code look pretty) as the very last commit? Why the very last? Because the commit will modify 50% of the file, and will make it hard for me to see the other changes. I have Prettier set to run every time I save the file, command+s. It keeps the code looking pretty, and easy for maintainers like myself to see what the changes are.

Reply:
alright,  will format it with prettier